src/core/model_event_queue.jl

```diff
@@ -215,18 +223,12 @@ function EventQueueABM(
     )
 end
 
+function Base.getindex(model::EventQueueABM{S, A, C}, id::Int) where {S,A,C<:StructVector}
+    return AgentWrapperSoA{A}(agent_container(model), id)
+end
```

Comment:
Why do we have this function? `getindex` should be defined only once at the abstract intertface level. It should call the `agent_container` function. A new internal function `retrieve_agent(container, id)` should be defined. This way `getindex` doesn't have to be implemented twice for the different model types.

Again, we have to get the interfaces good with this PR because in the future there will be more model types beyond event que and discrete time.